src/game-mechanics.js

```diff
@@ -1,59 +1,89 @@
-export const winStates = [
-  [
-    [0, 0],
-    [1, 0],
-    [2, 0],
-  ],
-  [
-    [0, 1],
-    [1, 1],
-    [2, 1],
-  ],
-  [
-    [0, 0],
-    [1, 0],
-    [2, 0],
-  ],
-  [
-    [0, 2],
-    [1, 2],
-    [2, 2],
-  ],
-  [
-    [0, 2],
-    [1, 1],
-    [2, 0],
-  ],
-  [
-    [0, 0],
-    [1, 1],
-    [2, 2],
-  ],
-];
+export const makeBoard = (size) =>
+  letIn(makeArrayOf(size), (makeArrayOfSize) =>
+    compose(makeArrayOfSize, makeArrayOfSize)("")
+  );
+
+export const playerHasWin = (player) => (board) =>
+  checkBoardForWin(board) === player;
+
+export const checkBoardForWin = (board) =>
+  compose(checkPlayerStateForWin(board.length), boardToPlayerState);
+
+const checkPlayerStateForWin = (boardSize) => (playerState) =>
+  Object.keys(playerState).reduce(
+    (winner, player) =>
+      checkPlayerState(boardSize)(playerState[player]) ? player : winner,
+    undefined
+  );
 
-export const validateWin = (gameState) => {
-  return winStates.filter((winState) => {
-    console.log(
-      gameState,
-      gameState[winState[0][0]][winState[0][1]] ===
-        gameState[winState[1][0]][winState[1][1]] &&
-        gameState[winState[1][0]][winState[1][1]] ===
-          gameState[winState[2][0]][winState[2][1]]
-    );
+const checkPlayerState = (boardSize) => ({ column, row }) =>
+  letIn({ eqBoardSize: eq(boardSize), gt0: gt(0) }, ({ eqBoardSize, gt0 }) =>
+    letIn(
+      {
+        columnStraight: column.some(eqBoardSize),
+        rowStraight: row.some(eqBoardSize),
+        diagonalStraight: column.all(gt0) && row.all(gt0),
+      },
+      (columnStraight, rowStraight, diagonalStraight) =>
+        columnStraight || rowStraight || diagonalStraight
+    )
+  );
 
-    return (
-      " " !== gameState[winState[0][0]][winState[0][1]] &&
-      gameState[winState[0][0]][winState[0][1]] ===
-        gameState[winState[1][0]][winState[1][1]] &&
-      gameState[winState[1][0]][winState[1][1]] ===
-        gameState[winState[2][0]][winState[2][1]]
-    );
-  });
-};
+/* Player State looks something like this, but can accept any string as a name
+ *
+ * {
+ *  x: { column: [0, 0, 0]
```

Comment:
Feels more naturals as tuples to me. Less complex to parse and easier to read the consequential computation.

Reply:
I like naming them because I always get column and row mixed up when I use tuples.